cpu/lm4f120/periph/uart.c

```diff
@@ -31,120 +31,217 @@
  */
 static uart_isr_ctx_t config[UART_NUMOF];
 
-static int init_base(uart_t uart, uint32_t baudrate);
+typedef struct {
+    unsigned long uart_sysctl;         /**< UART device in sysctl */
+    unsigned long uart_base;           /**< UART base address */
+    IRQn_Type uart_irq_chan;           /**< UART IRQ channel */
+    volatile unsigned long * uart_im_r;/**< UART RX Enable Register */
+
+    unsigned long gpio_sysctl;         /**< GPIO device in sysctl */
+    unsigned long gpio_port;           /**< GPIO port */
+    unsigned long rx;              /**< pin used for RX */
+    unsigned long tx;              /**< pin used for TX */
+    unsigned long mask_rx;            /**< Pin mask for RX*/
+    unsigned long mask_tx;            /**< Pin mask for TX*/
+} uart_conf_fixed_t;
+
+static const uart_conf_fixed_t ucf[] = {
```

Comment:
Regarding your question: I think allocating all these possible configuration parameters in the ROM is a little overkill and waste of resources. I personally would prefer to allocated the actual needed values in the board config (as done for the SPI driver of this CPU).

One thing we could consider is to reduce the needed values by automatically deducting some of them from others (though I did not check where/if this is possible here...). So getting `SYSCTL_PERIPH_UART0` from the `uart_num` or from `irq_num` could e.g. be a candidate